### PR TITLE
GH-1119: when req.clientClosed is true, set res statusCode to 0

### PR DESCRIPTION
**Pull request by DonutEspresso**
@micahr @yunong

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -861,7 +865,7 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
                     // in the case of formatter error, emit a formatterError
                     // event, which is like an uncaughtException scenario in
                     // that a response must be flushed by the handler.
-                    res.statusCode = 500;
+                    res.status(500);
```
Comment (DonutEspresso) [inline lib/server.js]: Just normalizing usage here.

**Comment by micahr** [comment]
👍 💯

**Comment by DonutEspresso** [comment]
FYI, this will now cause audit logs to look like this:

```sh
[2016-06-02T20:51:59.984Z]  INFO: server/23696 on foobarMachine handled: 0 (req_id=efab5fe6-6211-456d-9acb-544a0733a55a, audit=true, latency=2008, _audit=true, req.query="", req.version=*, req.clientClosed=true, res.headers=null)
    GET / HTTP/1.1
    host: localhost:3000
    user-agent: curl/7.43.0
    accept: */*
    --
    HTTP/1.1 0 undefined
    --
    req.timers: {
      "handler-0": 2004451
    }
```

**Comment by DonutEspresso** [comment]
Fixes #1119

**Comment by yunong** [comment]
LGTM

**Pull request merged**